lib/swaps/SwapClient.ts

```diff
@@ -253,6 +299,9 @@ abstract class SwapClient extends EventEmitter {
     if (!this.reconnectionTimer) {
       this.reconnectionTimer = setTimeout(this.reconnectionTimerCallback, SwapClient.RECONNECT_INTERVAL);
     }
+    if (!this.balanceAlertTimer) {
```

Comment:
This is an odd approach to me, it looks like we're setting a timer or every 30 seconds and then checking balance and sending out alerts, but that's not a very generic approach and I think would quickly get messy and hard to follow. I also dont' think it makes sense that SwapClientManager be so tightly coupled with alerts, I figure this sort of thing should go in Service (since it's entirely related to our user facing rpc) and it may even make sense to have another file/class to handle alerts.

If we indeed want a minimum amount of time between alerts of the same type (I agree that would be nice, although it seems that @kilrau disagreed above) then I'd rather we keep track of the timestamps of when we last sent out each type of alert, then before sending the alert we can compare the current time to the previous time and see if it exceeds the min threshold. 

The problem with this approach is that if we want another alert unrelated to balances that doesn't go on a 30 second timer, we'd either need to create a separate timer or come up with some different approach. A timer for checking whether we need to send an alert doesn't seem great to me in the first place, I'd expect we detect the alert condition in our regular flow and then raise an event.